geniza/corpus/views.py

```diff
@@ -646,6 +651,74 @@ def form_valid(self, form):
         return super(DocumentMerge, self).form_valid(form)
 
 
+class SourceAutocompleteView(PermissionRequiredMixin, autocomplete.Select2QuerySetView):
+    permission_required = ("corpus.change_document",)
+
+    def get_queryset(self):
+        """sources filtered by entered query, or all sources, ordered by author last name"""
+        q = self.request.GET.get("q", None)
+        qs = Source.objects.all().order_by("authors__last_name")
+        if q:
+            qs = qs.filter(
+                Q(title__icontains=q)
+                | Q(authors__first_name__istartswith=q)
+                | Q(authors__last_name__istartswith=q)
+                | Q(year__istartswith=q)
+                | Q(journal__icontains=q)
+                | Q(notes__icontains=q)
+                | Q(other_info__icontains=q)
+                | Q(languages__name__icontains=q)
+                | Q(volume__icontains=q)
+            ).distinct()
```

Comment:
This is really slow for me in the autocomplete. I wonder if we need to narrow the list of fields, or add database indexing?

FYI, in dark mode the autocomplete looks like it's only returning one result; I think the non-highlighted results are white on white.